### DISCUSSION TITLE
Devops Tools Error

### DISCUSSION THREAD

**Discussion by Naman1729** [Cyfrin/foundry-full-course-cu#242] [Q&A] [answered]
### Devops Tools Error
I have successfully deployed Raffle , and it was also displayed in broadcast. However, when I attempt to retrieve the most recent deployment using the DevOps tool, I encounter an error.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112473210/64b43098-776b-4d05-984b-b2d4bdf0907a)

**Comment by ghost** [comment]
Hello @Naman1729, most likely this error occured because you don't have `jq` installed, check the selected solution in discussion [#169](https://github.com/Cyfrin/foundry-full-course-f23/discussions/169#discussioncomment-6282434).

**Reply by shikhar229169** [reply]
I got the same error, I tried to install jq with ```sudo apt install jq```, but it throws the error:
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package jq is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'jq' has no installation candidate
```

**Reply by Shiva-Sai-ssb** [reply]
Hey @shikhar229169, try running this command 

```
sudo apt-get update && sudo apt-get install jq
```

**Reply by shikhar229169** [reply]
I resolved it. Actually there were some deb links in ```/etc/apt/sources.list``` which were commented, I uncommented them and it worked.

**Reply by Naman1729** [reply]
Hey @pacelliv , After installing jq, I encountered an error during the verification process. Instead of verifying **Raffle.sol** as intended, it attempted to verify **DevOpsTools.sol**.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112473210/c52a3ef2-6816-4eb9-9629-51948ffda080)

 Upon further investigation, I discovered that **Raffle.sol** was deployed first, followed by **DevOpsTools.sol**. Consequently, the verification process prioritized the most recently deployed contract, resulting in **Raffle.sol** not being verified.
 
 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112473210/60622089-b121-4cba-8125-a2bb29840de9)

**Reply by ghost** [reply, marked as answer]
Oof verifying has been problematic.

First try running `foundryup` and try again, hopefully it will help. 

If the issue persist (which wouldn't surprise me) run the verify command:

```bash
forge verify-contract --chain-id 11155111 --constructor-args $(cast abi-encode "constructor(uint64,bytes32,uint256,uint32,address)" <subscription_id> <gas_lane> <interval> <entrance_fee> <callback_gas_limit> <vrf_coordinator_address>) --etherscan-api-key <api_key> <contract_address> src/Raffle.sol:Raffle
```
For the entrance fee pass the value in wei, not "0.01 ether".
